### PR TITLE
Scripts: Adding flag that allows attaching tests to existing running application

### PR DESCRIPTION
**Pull request by RomanYarik**

### CODE REVIEW

File: packages/test-kit/src/attached-app.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    hostname: this.hostname,
+                    path,
+                    port: this.port,
+                    headers: {
+                        'Content-type': 'application/json'
+                    }
+                },
+                res => {
+                    res.on('data', () => {
+                        /**/
```
Comment (AviVahl) [inline packages/test-kit/src/attached-app.ts]: clarify why data is ignored

File: packages/test-kit/src/with-feature.ts [hunk 2]
```diff
@@ -13,11 +15,12 @@ export interface IFeatureTestOptions extends puppeteer.LaunchOptions {
     queryParams?: Record<string, string>;
     allowErrors?: boolean;
     runOptions?: Record<string, string>;
+    attach?: number;
```
Comment (AviVahl) [inline packages/test-kit/src/with-feature.ts]: add jsdoc. rename for clarity.

File: packages/test-kit/src/attached-app.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { IFeatureTarget } from '@wixc3/engine-scripts';
+import { request } from 'http';
+import { IExecutableApplication } from './types';
+
+const NODE_ENV_PATH = '/node-env';
+
+interface IEnvironmentHttpCall {
+    method: 'PUT' | 'DELETE' | 'GET';
+    path?: string;
```
Comment (AviVahl) [inline packages/test-kit/src/attached-app.ts]: make it required

File: packages/test-kit/src/attached-app.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    featureTarget?: IFeatureTarget;
+}
+
+export class AttachedApp implements IExecutableApplication {
+    constructor(private port = 3000, private hostname = 'localhost') {}
```
Comment (AviVahl) [inline packages/test-kit/src/attached-app.ts]: remove default 3000

**Comment by RomanYarik** [comment]
approved by Avi

**Pull request merged**